### PR TITLE
Add an option to let the user choose the qt style used by the app

### LINKED ISSUE #2662: Does KVIrc have a dark mode on Windows?
Hi everyone.
Is it possible on Windows to have the KVIrc window and all its elements displayed in dark colors (not only the channel/chat window contents)?
Thanks in advance!

### PR DESCRIPTION
**Pull request by ctrlaltca**
Fix #2662

### CODE REVIEW

**Comment by DarthGandalf** [comment]
So the -style command line flag will be ignored?

Any way to only change the default, but still honor the explicit user's config?

**Comment by ctrlaltca** [comment]
> So the -style command line flag will be ignored?

Unfortunately yes.

> Any way to only change the default, but still honor the explicit user's config?

Should we add an option? This way we could also remove the hardcoded style for gtk3 (#2117).
Hopefully we can get the style from options before the mainwindow get shown to avoid a sudden style change

**Comment by AbraxasDMG** [comment]
Is the command line switch the only way to apply different style at the moment (no possibility to do it through some config file or environment var)?

If so, KVIrc should also be able to recognize -style cmdline argument and avoid treating it as a server name to which it should connect automatically.

Just realized that this can be prevented if the style argument is provided as follows.
`-style=Fusion`

Then a new standalone server is not added to the list and automatically attempted to connect to.

**Comment by ctrlaltca** [comment]
@DarthGandalf please have a look at the second iteration of this patch, I've changed approach adding a new option in theme->general

@AbraxasDMG once we get this working you won't need to provide the commandline parameter anymore

File: src/modules/options/OptionsWidget_theme.cpp [hunk 1]
```diff
@@ -44,12 +46,38 @@ OptionsWidget_theme::OptionsWidget_theme(QWidget * parent)
 	    KVI_OPTION_BOOL(KviOption_boolUseGlobalApplicationFont));
 	connect(b, SIGNAL(toggled(bool)), f, SLOT(setEnabled(bool)));
 
-	addRowSpacer(0, 3, 1, 3);
+	addLabel(0, 3, 0, 3, __tr2qs_ctx("Qt Style:", "options"));
+	m_pQtStyle = new QComboBox(this);
+	addWidgetToLayout(m_pQtStyle, 1, 3, 1, 3);
+	m_pQtStyle->addItem(DEFAULT_QT_STYLE);
+	foreach(QString key, QStyleFactory::keys()) {
```
Comment (DarthGandalf) [inline src/modules/options/OptionsWidget_theme.cpp]: From Qt docs: New code should prefer C++11 range-based loops.
```suggestion
	for (const QString& key : QStyleFactory::keys()) {
```

**Comment by DarthGandalf** [comment]
Haven't tested, but seems fine, I guess

**Pull request merged**